transports/httpclient/src/main/java/org/commonjava/maven/galley/transport/htcli/ContentsFilteringTransferDecorator.java

```diff
@@ -63,7 +65,9 @@ public OverriddenBooleanValue decorateExists( final Transfer transfer )
                 final String path = transfer.getPath();
                 // pattern for "groupId path/(artifactId)/(version)/(filename)"
                 // where the filename starts with artifactId-version and is followed by - or .
-                final Pattern pattern = Pattern.compile( ".*/([^/]+)/([^/]+)/(\\1-\\2[-.][^/]+)$" );
+//                final Pattern pattern = Pattern.compile( ".*/([^/]+)/([^/]+)/(\\1-\\2[-.][^/]+)$" );
+                // NOS-1434 all files with snapshot version(in snapshot folder) should be ignored if !allowsSnapshots
+                final Pattern pattern = Pattern.compile( ".*/([^/]+)/([^/]+)/(.[^/]+)$" );
```

Comment:
Why is this pattern changed. Before it was designed to match only artifact, i.e. files matching pattern "<artifactId>-<version>" followed with a classifier or a file extension. This will match also any metadata and files not even on a GAV directory level that we don't really want to filter out. Or?